### ISSUE TITLE
Update compiler.py based on solcx 0.4.0 changes

### ISSUE THREAD

**Issue by ryu9827** [HyperLink-Technology/brownie#108]
Hello, I am the smart contract auditor from Blockchainlabs.nz auditing your SFT protocol contracts at the moment. We followed the readme.md in your repo, running `brownie test` in the repo folder, but got some errors.

* `brownie` Version: 1.0.0b3
* `solc` Version: 0.4.24, 0.4.25
* Python Version: 3.7
* OS: osx

### What was wrong?
The error message we got 
```
➜ brownie test                    
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/bin/brownie", line 11, in <module>
    load_entry_point('eth-brownie==1.0.0b2', 'console_scripts', 'brownie')()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pkg_resources/__init__.py", line 489, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2793, in load_entry_point
    return ep.load()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2411, in load
    return self.resolve()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2417, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/eth_brownie-1.0.0b2-py3.7.egg/brownie/__init__.py", line 3, in <module>
    from brownie.network import (
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/eth_brownie-1.0.0b2-py3.7.egg/brownie/network/__init__.py", line 6, in <module>
    from .rpc import Rpc as _Rpc
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/eth_brownie-1.0.0b2-py3.7.egg/brownie/network/rpc.py", line 11, in <module>
    from .account import Accounts
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/eth_brownie-1.0.0b2-py3.7.egg/brownie/network/account.py", line 14, in <module>
    from brownie.network.transaction import TransactionReceipt
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/eth_brownie-1.0.0b2-py3.7.egg/brownie/network/transaction.py", line 14, in <module>
    from brownie.project.sources import Sources
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/eth_brownie-1.0.0b2-py3.7.egg/brownie/project/__init__.py", line 9, in <module>
    from .build import Build
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/eth_brownie-1.0.0b2-py3.7.egg/brownie/project/build.py", line 10, in <module>
    from . import compiler
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/eth_brownie-1.0.0b2-py3.7.egg/brownie/project/compiler.py", line 6, in <module>
    import solcx
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/solcx/__init__.py", line 31, in <module>
    set_solc_version(get_installed_solc_versions()[-1])
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/solcx/install.py", line 90, in set_solc_version
    version = _check_version(version)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/solcx/install.py", line 214, in _check_version
    raise ValueError("solc version must be in the format v0.x.x")
ValueError: solc version must be in the format v0.x.x
```
Any command we run start with `brownie` gets error align to this.

* what command I have ran:
```
pip install eth-brownie
```
* the code that caused the failure
SFT, 29 April 2019, at commit de6454dd5659c4aa7ecba0a90e281394f3207a09

Thanks for help in advance.

**Comment by iamdefinitelyahuman** [comment]
Hi..  sorry about that.  Seems to be an issue with ``py-solc-x``.

What is the file structure in the ``/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/solcx/bin`` folder?

**Comment by ryu9827** [comment]
No problem. Below is the file structure under the path you mentioned:
<img width="945" alt="WeChatb5cc2fc185101d3e4decc5390a45bfb1" src="https://user-images.githubusercontent.com/24850827/57261753-3f060600-70bd-11e9-92d6-6bac6d233864.png">

**Comment by iamdefinitelyahuman** [comment]
The issue is that `py-solc-x` was interrupted while installing v0.5.8, and since it installs in place it left some files behind that it doesn't expect.   I'll update it to perform installs into a temp folder and copy them over once they're done, that should prevent this from happening in the future.

The simplest fix is to delete the `solcx/bin` folder and then reload Brownie.  You'll have to wait a bit while it compiles solc, unfortunately there are no precompiled binaries for OSX and the build process is.. slow. If you've installed any versions via brew they'll be imported to save time.

**Comment by iamdefinitelyahuman** [comment]
Discussed updates are [here](https://github.com/iamdefinitelyahuman/py-solc-x/tree/v0.3.1) but it's 5am for me, won't be headed to master until tomorrow.

**Comment by ryu9827** [comment]
One of the team members made it work by running these commands:
```
brew update && brew reinstall ethereum
brew install https://raw.githubusercontent.com/ethereum/homebrew-ethereum/master/solidity@4
```
Thanks for the help!

**Comment by iamdefinitelyahuman** [comment]
Glad it's working.

I just released [v1.0.0b4](https://github.com/HyperLink-Technology/brownie/releases/tag/v1.0.0b4) which includes the updated `py-solc-x` and should be more friendly to OSX.
